python.md

```diff
@@ -974,6 +974,49 @@ from path.to.some.module import \
     thing1, thing2, thing3, thing4
 ```
 
+### Make function signatures explicit
+
+Avoid using ``**kwargs`` in function signatures without good reason. Code with loosely defined
+function signatures can be difficult to work with, as it's unclear what variables are
+entering the function.
+ 
+```python
+def do_something(**kwargs):  # Don't do this
+   ...
+```
+
+Be explicit instead:
+
+```python
+def do_something(foo: int, bar: str):
+   ...
+```
+
+This includes functions that wrap lower level functionality, such as model creation methods:
```

Comment:
I agree with trying to get rid of `**kwargs` in principle, but there are a fair few points I've seen in the codebase which end up with you passing 8-10 variables all the way through a stack to get to where they are actually needed.

Obviously that is a bad code smell, but would it be worth this section offering other ways to avoid that (when using `**kwargs` would be the easiest one). I'm thinking of things like passing around data objects etc.

Reply:
> I've seen in the codebase which end up with you passing 8-10 variables all the way through a stack to get to where they are actually needed.

TBH, I prefer that to having `**kwargs` used for a chain of calls. As you say though, there's probably a better way.

Reply:
Thanks @leamingrad, this gets to the heart of the problem. I've added an additional commit to mention this:  https://github.com/octoenergy/conventions/pull/66/commits/8801c37cce5d65d5ac2ec7e3108341e630a33bbc